slave.mk

```diff
@@ -1192,6 +1194,68 @@ $(DOCKER_LOAD_TARGETS) : $(TARGET_PATH)/%.gz-load : .platform docker-start $$(TA
 ## Installers
 ###############################################################################
 
+ifeq ($(ENABLE_RFS_SPLIT_BUILD),y)
+define rfs_get_installer_dependencies
+$(1)__$($(1)_MACHINE)__rfs.squashfs $(if $($(1)_DEPENDENT_MACHINE),$(1)__$($(1)_DEPENDENT_MACHINE)__rfs.squashfs)
+endef
+
+$(foreach installer, $(SONIC_INSTALLERS), $(eval $(installer)_RFS_DEPENDS=$(call rfs_get_installer_dependencies,$(installer))))
+
+SONIC_RFS_TARGES= $(foreach installer, $(SONIC_INSTALLERS), $(call rfs_get_installer_dependencies,$(installer)))
+SONIC_TARGET_LIST+= $(addprefix $(TARGET_PATH)/, $(SONIC_RFS_TARGES))
+endif
+
+$(addprefix $(TARGET_PATH)/, $(SONIC_RFS_TARGES)) : $(TARGET_PATH)/% : \
+        .platform \
+        build_debian.sh \
+        $(addprefix $(IMAGE_DISTRO_DEBS_PATH)/,$(INITRAMFS_TOOLS) $(LINUX_KERNEL))
+	$(HEADER)
+
+	$(eval installer=$(word 1,$(subst __, ,$*)))
+	$(eval machine=$(word 2,$(subst __, ,$*)))
+
+	echo 'RFS first stage: building $* (installer: $(installer), machine: $(machine))'
+
+	export debs_path="$(IMAGE_DISTRO_DEBS_PATH)"
+	export initramfs_tools="$(IMAGE_DISTRO_DEBS_PATH)/$(INITRAMFS_TOOLS)"
+	export linux_kernel="$(IMAGE_DISTRO_DEBS_PATH)/$(LINUX_KERNEL)"
+	export kversion="$(KVERSION)"
+	export image_type="$($(installer)_IMAGE_TYPE)"
+	export sonicadmin_user="$(USERNAME)"
+	export sonic_asic_platform="$(patsubst %-$(CONFIGURED_ARCH),%,$(CONFIGURED_PLATFORM))"
+	export RFS_SPLIT_BUILD_START=y
+
+	j2 -f env files/initramfs-tools/union-mount.j2 onie-image.conf > files/initramfs-tools/union-mount
+	j2 -f env files/initramfs-tools/arista-convertfs.j2 onie-image.conf > files/initramfs-tools/arista-convertfs
+
+	RFS_SQUASHFS_NAME=$* \
+	DEBUG_IMG="$(INSTALL_DEBUG_TOOLS)" \
```

Comment:
Env vars such as 

- DEBUG_IMG
- SONIC_ENFORCE_VERSIONS
- MASTER_*_VERSION 

are not used in first stage, can they be removed?

Reply:
Good point, I'll remove the vars that are not needed